src/logging.cc

```diff
@@ -476,6 +494,9 @@ class LogFileObject : public base::Logger {
   unsigned int rollover_attempt_;
   int64 next_flush_time_;         // cycle count at which to flush log
   WallTime start_time_;
+  std::list<Filetime> file_list_;
```

Comment:
What's the reason for choosing a `std::list` here? Looks like you want a `std::priority_queue` here since you are constantly sorting the container elements.

Also the member variable name should indicate what kind of times are stored.

Reply:
Thanks for review, it is because of need to access the last element @ https://github.com/google/glog/blob/0f7ffec4ac55a5513ab10d85d2d51ac60e6bd4a9/src/logging.cc#L1252

Reply:
That's not a good reason for choosing `std::list`. Given you are continually sorting, `std::priority_queue` is probably a better choice.

Reply:
Maybe `std::set` is a better choice? It is ordered and have access to first and last element.

Reply:
No, `std::set` is only useful if you lookup specific elements which you don't. The container also doesn't really give direct access to the top element. This is what `std::priority_queue` is for.